tests/test_graphql.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+from graphql import GraphQLError
+from graphql.validation.rules import ValidationRule
+
+from ariadne import graphql, graphql_sync, subscribe
+
+
+class AlwaysInvalid(ValidationRule):
+    def leave_operation_definition(  # pylint: disable=unused-argument
+        self, *args, **kwargs
+    ):
+        self.context.report_error(GraphQLError("Invalid"))
+
+
+def test_graphql_sync_works(schema):
+    success, result = graphql_sync(
+        schema, {"query": '{ hello(name: "world") }'}
+    )
+    assert success
+    assert result["data"] == {"hello": "Hello, world!"}
+
+
+def test_graphql_sync_uses_validation_rules(schema):
+    success, result = graphql_sync(
+        schema, {"query": '{ hello(name: "world") }'}, validation_rules=[AlwaysInvalid]
+    )
+    assert not success
+    assert result["errors"][0]["message"] == "Invalid"
+
+
+@pytest.mark.asyncio
+async def test_graphql_works(schema):
```

Comment:
```suggestion
def test_graphql_executes_query(schema):
```

Reply:
This has to be a coroutine.

Reply:
ah, right. I've just copied over previous comment and edited.